packages/core/src/rules/common/spec.ts

```diff
@@ -114,8 +114,27 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
           propValue = resolve(propValue).node;
         }
 
+        if (propSchema.items && propSchema.items?.enum) {
+          if (Array.isArray(propValue)) {
+            propValue.forEach((item) => {
+              if (!propSchema.items?.enum?.includes(item !== null ? item : JSON.stringify(item))) {
+                report({
+                  location: propLocation,
+                  message: `\`${propName}\` can be one of the following only: ${propSchema.items
+                    ?.enum!.map((i) => `"${i}"`)
+                    .join(', ')}.`,
+                  from: refLocation,
+                  suggest: getSuggest(item, propSchema.items?.enum as string[]),
+                });
+              }
+            });
+          }
+        }
+
         if (propSchema.enum) {
-          if (!propSchema.enum.includes(propValue)) {
+          if (
+            !propSchema.enum.includes(propValue !== null ? propValue : JSON.stringify(propValue))
+          ) {
```

Comment:
I believe we have a duplication here. Can we avoid it?

Reply:
I moved similar code to separate function